MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/adapter/ItemAdapter.java

```diff
@@ -38,8 +53,42 @@ public ItemAdapterViewHolder onCreateViewHolder(ViewGroup viewGroup, int index)
 
     @Override
     public void onBindViewHolder(ItemAdapterViewHolder itemAdapterViewHolder, int i) {
-        DataSource sharedDataSource = MediaTrackerApplication.getSharedDataSource();
-        itemAdapterViewHolder.update(sharedDataSource.getAllBooks().get(i), listener);
+        List<Book> bookList = new ArrayList<Book>();
+        bookList = MediaTrackerApplication.getSharedDataSource().getAllBooks();
+
+        if (currentSortCriteria == "title") {
```

Comment:
If you make the `currentSortCriteria` an `enum`, this would turn into:

```java
switch (currentSortCriteria) {
   case TITLE:
      //sort my title as you do with comparator
      break;
   case AUTHOR:
      //sort by author as you do with comparator
      break;
   case ADDED:
      // sort by added as you do with comparator
      break;
   default: 
      throw new IllegalArgumentException("Cannot sort by: " + currentSortCriteria);
}
```

This way you keep the strong typing by the enum, and not rely on the strings. If by mistake you had misspelled `title` here, it would have been a little bit hard to figure out what's wrong. With the enum we avoid this.